### PR TITLE
aws - rds support multi-az start/stop

### PR DESCRIPTION
**Pull request by kapilt**
addresses #3109

### CODE REVIEW

File: c7n/resources/rds.py [hunk 1]
```diff
@@ -475,11 +475,17 @@ def process_tag_removal(self, client, resource, candidates):
...
+    # Note that this doesn't really specify what happens for all the nosql engines
+    # that are available as rds engines.
     if db.get('DBInstanceStatus') != state:
         return False
 
-    if db.get('MultiAZ'):
+    if db.get('MultiAZ') and db['Engine'].startswith('sqlserver-'):
+        return False
+
+    if db['Engine'] in ('docdb', 'neptune'):
```
Comment (thisisshi) [inline c7n/resources/rds.py]: imo would be better as a whitelist as that's what the docs provide, e.g. 
```
db['Engine'] not in ('postgresql', 'mariadb'...)
```

**Comment by thisisshi** [comment]
looks like the fake resource is missing the `Engine` key, and the default `state` arg in _eligible_start_stop seems suspect but otherwise lgtm

File: c7n/resources/rds.py (hunk 1 above)
Comment (kapilt) [reply c7n/resources/rds.py]: okay.. fwiw the full set of engines is {'postgres', 'sqlserver-ee', 'oracle-se2', 'mariadb', 'docdb', 'oracle-ee', 'sqlserver-ex', 'sqlserver-se', 'oracle-se', 'aurora-mysql', 'mysql', 'oracle-se1', 'aurora-postgresql', 'sqlserver-web', 'aurora', 'neptune'}

Comment (kapilt) [reply c7n/resources/rds.py]: ie this applies to every single rdbms rds engine

**Comment by kapilt** [comment]
why does the pre-existing default state arg seem suspect to you?

**Comment by thisisshi** [comment]
calling start before stop would overwrite the default state argument to "stopped" and return invalid results

**Comment by kapilt** [comment]
strings are immutable in python, ie the default state argument isn't mutable. its always going to be state='available' unless another value is passed when invoking the function. default arguments are problematic when they reference a mutable object like a list or dict, because in those cases any modifications to the arg persist across invocations. ie. afaics this usage is fine, but perhaps i'm misunderstanding your concern.

**Comment by thisisshi** [comment]
gotcha, thanks misunderstanding on my part

**Pull request merged**